src/core/core.scale.js

```diff
@@ -287,8 +288,15 @@ module.exports = Element.extend({
 		helpers.callback(this.options.beforeBuildTicks, [this]);
 	},
 	buildTicks: helpers.noop,
-	afterBuildTicks: function() {
-		helpers.callback(this.options.afterBuildTicks, [this]);
+	afterBuildTicks: function(ticks) {
+		var me = this;
+		// ticks is empty for old axis implementations here
+		if (ticks.length) {
+			return helpers.callback(me.options.afterBuildTicks, [me, ticks]);
+		}
+		// Support old implementatios (that modified `this.ticks` directly in buildTicks)
+		me.ticks = helpers.callback(me.options.afterBuildTicks, [me, me.ticks]) || me.ticks;
```

Comment:
Because next we call `labels = me.convertTicksToLabels(ticks) || me.ticks;`
But the default implementation ignores arguments and does not return:
```js
	convertTicksToLabels: function() {
		var me = this;
		// Convert ticks to strings
		var tickOpts = me.options.ticks;
		me.ticks = me.ticks.map(tickOpts.userCallback || tickOpts.callback, this);
	},
```